MusicNotationKit/MusicNotationKit/Measure.swift

```diff
@@ -53,50 +55,327 @@ public struct Measure: ImmutableMeasure, Equatable {
+        try insert(noteCollections, at: index, inSet: setIndex, skipTies: false)
+    }
+
+    public mutating func removeNote(at index: Int, inSet setIndex: Int = 0) throws {
+        try removeNote(at: index, inSet: setIndex, skipTies: false)
+    }
 
+    public mutating func removeNotesInRange(_ indexRange: CountableClosedRange<Int>, inSet setIndex: Int = 0) throws {
+        try removeNotesInRange(indexRange, inSet: setIndex, skipTies: false)
+    }
+
+    /// Create a `Tuplet` from a note range. See `Tuplet.init` for more details.
+    /// This function also makes sure that the `noteRange` does not start or end
+    /// across a `Tuplet` boundary.
+    public mutating func createTuplet(_ count: Int, _ baseNoteDuration: NoteDuration, inSpaceOf baseCount: Int? = nil, fromNotesInRange noteRange: CountableClosedRange<Int>, inSet setIndex: Int = 0) throws {
+        var newMeasure = self
+        let startCollectionIndex = try newMeasure.noteCollectionIndex(fromNoteIndex: noteRange.lowerBound, inSet: setIndex)
+        if startCollectionIndex.tupletIndex != nil {
+            guard let tupletIndex = startCollectionIndex.tupletIndex, tupletIndex == 0 else {
+                throw MeasureError.invalidTupletIndex
+            }
+        }
+
+        let endCollectionIndex = try newMeasure.noteCollectionIndex(fromNoteIndex: noteRange.upperBound, inSet: setIndex)
+        if endCollectionIndex.tupletIndex != nil {
+            guard let tupletIndex = endCollectionIndex.tupletIndex,
+                let tuplet = newMeasure.notes[setIndex][endCollectionIndex.noteIndex] as? Tuplet else {
+                    assertionFailure("note collection should be tuplet, but cast failed")
+                    throw MeasureError.internalError
+            }
+            guard tuplet.noteCount == tupletIndex + 1 else {
+                throw MeasureError.invalidTupletIndex
+            }
+        }
+
+        var expectedCount = noteRange.upperBound - noteRange.lowerBound + 1
+        var tupletNotes = [NoteCollection]()
+        var index = startCollectionIndex.noteIndex
+
+        while expectedCount > 0 && notes[setIndex].count > index{
+            tupletNotes.append(notes[setIndex][index])
+            expectedCount -= notes[setIndex][index].noteCount
+            index += 1
+        }
+
+        let newTuplet = try Tuplet(count, baseNoteDuration, inSpaceOf: baseCount, notes: tupletNotes)
+        try newMeasure.removeNotesInRange(noteRange, inSet: setIndex)
+        try newMeasure.insert(newTuplet, at: noteRange.lowerBound, inSet: setIndex)
+        self = newMeasure
+    }
+
+    public mutating func breakdownTuplet(at index: Int, inSet setIndex: Int = 0) throws {
+        var newMeasure = self
+        let collectionIndex = try newMeasure.noteCollectionIndex(fromNoteIndex: index, inSet: setIndex)
+        guard let tupletIndex = collectionIndex.tupletIndex, tupletIndex == 0 else {
+            throw MeasureError.invalidTupletIndex
+        }
```

Comment:
Remove this check. You should be able to just pick anywhere within the tuplet to be able to break it down.